shared/api/auth.go

```diff
@@ -24,3 +24,140 @@ const (
 	// IdentityTypeOIDCClient represents an identity that authenticates with OIDC..
 	IdentityTypeOIDCClient = "OIDC client"
 )
+
+// Identity is the type for an authenticated party that can make requests to the HTTPS API.
+//
+// swagger:model
+type Identity struct {
+	// AuthenticationMethod is the authentication method that the identity
+	// authenticates to LXD with.
+	// Example: tls
+	AuthenticationMethod string `json:"authentication_method" yaml:"authentication_method"`
+
+	// Type is the type of identity.
+	// Example: oidc-service-account
+	Type string `json:"type" yaml:"type"`
+
+	// Identifier is a unique identifier for the identity (e.g. certificate fingerprint or email for OIDC).
+	// Example: jane.doe@example.com
+	Identifier string `json:"id" yaml:"id"`
+
+	// Name is the Name claim of the identity if authenticated via OIDC, or the name
+	// of the certificate if authenticated with TLS.
+	// Example: Jane Doe
+	Name string `json:"name" yaml:"name"`
+}
+
```

Comment:
So just to confirm there's no way to create an identity directly ATM, they must login directly or create a TLS trust entry?